packages/js-sdk/src/datastore/networkstore.ts

```diff
@@ -198,22 +198,28 @@ export class NetworkStore {
     const batchSize = 100;
     const apiVersion = getApiVersion();
 
-    if (apiVersion !== 5 && isArray(docs)) {
-      throw new KinveyError('Unable to create an array of entities. Please create entities one by one.');
+    if (apiVersion < 5 && isArray(docs)) {
+      throw new KinveyError('Unable to create an array of entities. Please create entities one by one or use API version 5 or newer.');
     }
 
     if (isArray(docs) && docs.length > batchSize) {
       let i = 0;
 
-      const batchCreate = async (createResults: any = []): Promise<any> => {
+      const batchCreate = async (previousResult: any = { entities: [], errors: [] }): Promise<any> => {
         if (i >= docs.length) {
-          return createResults;
+          return previousResult;
         }
 
         const batch = docs.slice(i, i + batchSize);
-        i += batchSize;
         const result = await this.create(batch, options);
-        return batchCreate(createResults.concat(result));
+        result.errors.forEach((e) => (e.index += i));
+        const combinedResult = {
+          entities: previousResult.entities.concat(result.entities),
```

Comment:
`docs.slice(i, i + batchSize)` would ideally be a loop, working with indices of the array, and not create slices off the array each time. Think it would be good to not add to this overhead by copying the entire results array for each batch. I mean `concat` creates a new array every time. I think it's better to do `previousResult.push(... result.entities)` - this way we only iterate over the batch, not all the results for every batch.